README.md

```diff
@@ -1 +1,133 @@
+as in:
+
+    $ref: "filename.json#/definitions/field`
+
+This is a by product of the library we use to evolve the error output. As this may change, best practice is for the
+`id` in the schema to exactly match the filename.  See the `derived.json` and `base.json` test schemas for an example.
+
+# Rejecting invalid REST payloads
+
+Although you usually will build some fault tolerance into your components, on some level they expect to deal with
+data that has the right structure, and which contains the expected type of information (strings, booleans, dates, etc).
+
+The `gpii.schema.middleware` component provided with this package rejects invalid payloads, which allows your server-side
+components to safely assume they will only receive JSON data in the correct format.
+
+The component is intended to be wired into an existing rest endpoint as in the following example:
+
+    var fluid = fluid || require("infusion");
+    var gpii  = fluid.registerNamespace("gpii");
+
+    require("gpii-express");
+    require("gpii-json-schema");
+
+    fluid.defaults("gpii.schema.tests.handler", {
+      gradeNames: ["gpii.express.handler"],
+      invokers: {
+        handleRequest: {
+          funcName: "{that}.sendResponse",
+          args:     [200, "Someone sent me a valid JSON payload."]
+        }
+      }
+    );
+
+    gpii.express({
+      gradeNames: ["gpii.express.requestAware.router"],
+      handlerGrades: ["gpii.schema.tests.handler"],
+      path: "/gatekeeper",
+      components: {
+        gatekeeper: {
+          type: "gpii.schema.middleware",
+          options: {
+            schemaContent: { "type": "object", "properties": { "required": { "type": "boolean" } }, "required": ["required"]}
+          }
+        }
+      }
+    });
+
+If you were to launch this example which is adapted from the tests in this package), you would have a rest endpoint
+(/gatekeeper) that rejects all invalid responses and outputs a success message when the JSON is valid.
+
+The component doesn't know anything about how you intend to use the data.  It only examines the payload and steps in if
+the payload is not valid according to the configured JSON Schema.  The JSON Schema language lets you combine and reuse
+part or all of an existing schema (with the caveats described above), so you should be able to describe a broad range
+of data you accept in a single JSON Schema.
+
+# Adding headers to REST responses
+
+When writing REST interfaces, we also commonly return JSON data in response to a request.  JSON Schemas can also be
+used to provide hints about the output format we are using.
+
+The working group that writes the JSON Schema standard [has outlined two approaches for labeling outgoing responses](http://json-schema.org/latest/json-schema-core.html#anchor33).
+Both of these involve setting HTTP headers in the outgoing response, as in:
+
+    Content-Type: application/my-media-type+json; profile="http://example.com/my-hyper-schema#"
```

Comment:
Too much coffee?  The hyper schema working group must have worked on the docs?  Anyway, we don't require any of the hyper-ness yet, I can update the example to avoid even an indirect suggestion that it is preferred.